E5_documentation.md

```diff
@@ -11,4 +14,45 @@ For doing so:
 - Check the documentation looks nice locally: you can generate the html files with ```make html``` (see requirements.txt).
 - push to a branch and perform a pull request to the main branch.
 
-After review and once the branch is merged into th main branch, the online documentation will be automatically updated.
\ No newline at end of file
+After review and once the branch is merged into th main branch, the online documentation will be automatically updated.
+
+## Documentation of all packages
+
+The software comprises several packages.
+These are the packages that are downloaded via treep when installing the software from source and using colcon as described [here](A1_overview_and_installation#via-colcon-workspace).
+Some of these package have documentation, which can be built locally
+
+### How to build
+
+Using colcon:
+
+```bash
+cd /path/to/Software
+cd workspace
+colcon build --cmake-args ' -DCMAKE_BUILD_TYPE=Release ' -DGENERATE_DOCUMENTATION=ON
+```
+
+For this to work, you may need to pip install some packages: m2r, mistune (version 0.8.4), recommonmark, sphinx, sphinxcontrib-moderncmakedomain, breath.
+
+Once compilation is successfully finished, html documentation can be found in paths:
```

Comment:
HTML is not generated automatically (at least on my system), I get some Sphinx files (restructured text) along a doxyfile (with GENERATE_HTML = No)

Reply:
@alexanderdittrich For all packages, or only a few ? For example, do you get a html documentation for shared memory ?